hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/DiskBalancerService.java

```diff
@@ -0,0 +1,767 @@
+    return builder.setIsRunning(this.operationalState == DiskBalancerOperationalState.RUNNING)
+        .setBalancedBytes(totalBalancedBytes.get())
+        .setDiskBalancerConf(
+            HddsProtos.DiskBalancerConfigurationProto.newBuilder()
+                .setThreshold(threshold)
+                .setDiskBandwidthInMB(bandwidthInMB)
+                .setParallelThread(parallelThread)
+                .setStopAfterDiskEven(stopAfterDiskEven)
+                .build())
+        .build();
+  }
+
+  @Override
+  public BackgroundTaskQueue getTasks() {
+    BackgroundTaskQueue queue = new BackgroundTaskQueue();
+
+    if (this.operationalState == DiskBalancerOperationalState.STOPPED ||
+        this.operationalState == DiskBalancerOperationalState.PAUSED_BY_NODE_STATE) {
+      return queue;
+    }
+    metrics.incrRunningLoopCount();
+
+    if (shouldDelay()) {
+      metrics.incrIdleLoopExceedsBandwidthCount();
+      return queue;
+    }
+
+    int availableTaskCount = parallelThread - inProgressContainers.size();
+    if (availableTaskCount <= 0) {
+      LOG.info("No available thread for disk balancer service. " +
+          "Current thread count is {}.", parallelThread);
+      return queue;
+    }
+
+    for (int i = 0; i < availableTaskCount; i++) {
+      Pair<HddsVolume, HddsVolume> pair = volumeChoosingPolicy
+          .chooseVolume(volumeSet, threshold, deltaSizes, containerDefaultSize);
+      if (pair == null) {
+        continue;
+      }
+      HddsVolume sourceVolume = pair.getLeft(), destVolume = pair.getRight();
+      ContainerData toBalanceContainer = containerChoosingPolicy
+          .chooseContainer(ozoneContainer, sourceVolume, inProgressContainers);
+      if (toBalanceContainer != null) {
+        DiskBalancerTask task = new DiskBalancerTask(toBalanceContainer, sourceVolume,
+            destVolume);
+        queue.add(task);
+        inProgressContainers.add(ContainerID.valueOf(toBalanceContainer.getContainerID()));
+        deltaSizes.put(sourceVolume, deltaSizes.getOrDefault(sourceVolume, 0L)
+            - toBalanceContainer.getBytesUsed());
+      } else {
+        // release destVolume committed bytes
+        destVolume.incCommittedBytes(0 - containerDefaultSize);
+      }
+    }
+
+    if (queue.isEmpty()) {
+      bytesToMove = 0;
+      if (stopAfterDiskEven) {
+        LOG.info("Disk balancer is stopped due to disk even as" +
+            " the property StopAfterDiskEven is set to true.");
+        this.operationalState = DiskBalancerOperationalState.STOPPED;
+        try {
+          // Persist the updated shouldRun status into the YAML file
+          writeDiskBalancerInfoTo(getDiskBalancerInfo(), diskBalancerInfoFile);
+        } catch (IOException e) {
+          LOG.warn("Failed to persist updated DiskBalancerInfo to file.", e);
+        }
+      }
+      metrics.incrIdleLoopNoAvailableVolumePairCount();
+    } else {
+      bytesToMove = calculateBytesToMove(volumeSet);
```

Comment:
It does not consider committed space for bytes to move, this may give wrong info, as some volume is mostly occupied as committed, and treated as source volume.